plugins/destination/postgresql/client/migrate.go

```diff
@@ -111,89 +115,103 @@ func (c *Client) listPgTables(ctx context.Context, pluginTables schema.Tables) (
-		case schema.TypeCIDRArray:
-			table.Columns[i].Type = schema.TypeInetArray
-		case schema.TypeMacAddr:
-			table.Columns[i].Type = schema.TypeString
-		case schema.TypeMacAddrArray:
-			table.Columns[i].Type = schema.TypeStringArray
-		}
-		if table.Columns[i].CreationOptions.PrimaryKey {
-			table.Columns[i].CreationOptions.NotNull = true
-		}
-	}
-	return table
-}
+// func (c *Client) normalizeTableCockroach(table *arrow.Schema) *arrow.Schema {
+// 	for i := range table.Fields() {
+// 		if !c.enabledPks() {
+// 			table.Columns[i].CreationOptions.PrimaryKey = false
+// 		}
+// 		switch table.Columns[i].Type {
+// 		case schema.TypeCIDR:
+// 			table.Columns[i].Type = schema.TypeInet
+// 		case schema.TypeCIDRArray:
+// 			table.Columns[i].Type = schema.TypeInetArray
+// 		case schema.TypeMacAddr:
+// 			table.Columns[i].Type = schema.TypeString
+// 		case schema.TypeMacAddrArray:
+// 			table.Columns[i].Type = schema.TypeStringArray
+// 		}
+// 		if table.Columns[i].CreationOptions.PrimaryKey {
+// 			table.Columns[i].CreationOptions.NotNull = true
+// 		}
+// 	}
+// 	return table
+// }
 
-func (c *Client) normalizeTablePg(table *schema.Table) *schema.Table {
-	for i := range table.Columns {
-		if !c.enabledPks() {
-			table.Columns[i].CreationOptions.PrimaryKey = false
+func (c *Client) normalizeTablePg(table *arrow.Schema, pgTable *arrow.Schema) *arrow.Schema {
+	fields := make([]arrow.Field, len(table.Fields()))
+	for i, f := range table.Fields() {
+		metadata := make(map[string]string, 0)
+		if !schema.IsPk(f) {
+			metadata[schema.MetadataPrimaryKey] = schema.MetadataFalse
 		}
-		if table.Columns[i].CreationOptions.PrimaryKey {
-			table.Columns[i].CreationOptions.NotNull = true
+		if c.enabledPks() && schema.IsPk(f) {
+			metadata[schema.MetadataPrimaryKey] = schema.MetadataTrue
+			f.Nullable = true
 		}
-		// this is for backward compatibility as I believe this is as of right now defined on the source
-		if c.enabledPks() && len(table.PrimaryKeys()) == 0 {
-			cqIdColumn := table.Columns.Get(schema.CqIDColumn.Name)
-			if cqIdColumn != nil {
-				cqIdColumn.CreationOptions.PrimaryKey = true
-			}
+		f.Metadata = arrow.MetadataFrom(metadata)
+		fields[i] = f
+	}
+	mdMap := make(map[string]string)
+	if pgTable != nil {
+		mdMap[schema.MetadataTableName] = schema.TableName(pgTable)
+		if constraintName, ok := pgTable.Metadata().GetValue(schema.MetadataConstraintName); ok {
+			mdMap[schema.MetadataConstraintName] = constraintName
 		}
 	}
-	return table
+	mdMap[schema.MetadataTableName] = schema.TableName(table)
+	md := arrow.MetadataFrom(mdMap)
+	return arrow.NewSchema(fields, &md)
 }
 
-func (c *Client) normalizeTable(table *schema.Table) *schema.Table {
+func (c *Client) normalizeTable(table *arrow.Schema, pgTable *arrow.Schema) *arrow.Schema {
 	switch c.pgType {
-	case pgTypeCockroachDB:
-		return c.normalizeTableCockroach(table)
+	// case pgTypeCockroachDB:
+	// 	return c.normalizeTableCockroach(table)
```

Comment:
Same here, we could delete this and even simplify to a single `if`

Reply:
Fixed.